easy/validparenthesis.py

```diff
@@ -0,0 +1,98 @@
+"""
+Given a string s containing just the characters '(', ')', '{', '}', '[' and ']',
+determine if the input string is valid.
+
+An input string is valid if:
+
+Open brackets must be closed by the same type of brackets.
+Open brackets must be closed in the correct order.
+
+Examples:
+
+Input: s = "()"
+Output: true
+
+Input: s = "()[]{}"
+Output: true
+
+Input: s = "(]"
+Output: false
+
+Input: s = "([)]"
+Output: false
+
+Input: s = "{[]}"
+Output: true
+"""
+
+import unittest
+
+class Solution:
+    def isValid(self, s):
+        stack = []
+        opening_chars = ["(", "{", "["]
```

Comment:
Could ditch `opening_chars` if you reversed the order of `pairs` and just used that. Would also be better for lookups (although with only three keys it really doesn't matter here.)